src/persist-client/src/impl/machine.rs

```diff
@@ -400,6 +404,26 @@ where
     }
 }
 
+pub async fn retry_determinate<R, F, WorkFn>(
```

Comment:
should we use the same exponential backoff as retry_external?

Reply:
oooooof yes absolutely. long story. very glad you caught this!